### ISSUE TITLE
Can't start display-manager.service on LiveCD

### ISSUE THREAD

**Issue by cstrahan** [NixOS/nixpkgs#10381]
When starting `display-manager.service` on a LiveCD, I get errors along the lines of:

```
Starting X11 ...
Started X11 Server.
Multiple occurrences of section [X-*-Core] in /nix/store/...-kdmrc. Consider merging them.
Multiple occurrences of section [Shutdown] in /nix/store/...-kdmrc. Consider merging them.
Multiple occurrences of section [X-*-Core] in /nix/store/...-kdmrc. Consider merging them.
Multiple occurrences of key 'Session' in section [X-*-Core] of /nix/store/...-kdmrc
Multiple occurrences of key 'AllowRootLogin' in section [X-*-Core] of /nix/store/...-kdmrc
Multiple occurrences of key 'SessionDirs' in section [X-*-Core] of /nix/store/...-kdmrc
Multiple occurrences of section [X-*-Greeter] in /nix/store/...-kdmrc. Consider merging them.
Multiple occurrences of section [X-:*-Core] in /nix/store/...-kdmrc. Consider merging them.
Multiple occurrences of key 'ServerCmd' in section [X-:*-Core] of /nix/store/...-kdmrc
X server died during startup
X server for display :0 cannot be started. session disabled
```

**Comment by cstrahan** [comment]
This is with the latest `nixos-unstable`: e9162732095.

Here's the contents of the `kdmrc`: https://gist.github.com/cstrahan/c4122fe2763eb7d32152

**Comment by cstrahan** [comment]
Ah, I thought it was the `kderc` at fault, but it's not.

Here's my `/var/log/X.0.log`: https://gist.github.com/cstrahan/35cdb292c7994e094f72

And (I think) these are the relevant bits:

    [    14.790] (II) modesetting: Driver for Modesetting Kernel Drivers: kms
    [    14.790] (++) using VT number 7
    
    [    14.797] (II) [KMS] drm report modesetting isn't supported.
    [    14.797] (WW) Falling back to old probe method for cirrus
    [    14.797] (WW) Falling back to old probe method for modesetting
    [    14.797] (EE) open /dev/dri/card0: No such file or directory
    [    14.797] (EE) Screen 0 deleted because of no matching config section.
    [    14.797] (II) UnloadModule: "radeon"
    [    14.797] (II) Loading sub module "vbe"
    [    14.797] (II) LoadModule: "vbe"
    [    14.800] (II) Loading /nix/store/w4nh4907ihcbm0szg6q5n56g5ija5bqh-xorg-server-1.17.2/lib/xorg/modules/libvbe.so
    [    14.801] (II) Module vbe: vendor="X.Org Foundation"
    [    14.801] 	compiled for 1.17.2, module version = 1.1.0
    [    14.801] 	ABI class: X.Org Video Driver, version 19.0
    [    14.801] (II) Loading sub module "int10"
    [    14.801] (II) LoadModule: "int10"
    [    14.805] (II) Loading /nix/store/w4nh4907ihcbm0szg6q5n56g5ija5bqh-xorg-server-1.17.2/lib/xorg/modules/libint10.so
    [    14.806] (II) Module int10: vendor="X.Org Foundation"
    [    14.806] 	compiled for 1.17.2, module version = 1.0.0
    [    14.806] 	ABI class: X.Org Video Driver, version 19.0
    [    14.806] (II) VESA(0): initializing int10
    [    14.807] (EE) VESA(0): V_BIOS address 0x43420 out of range
    [    14.807] (II) UnloadModule: "vesa"
    [    14.807] (II) UnloadSubModule: "int10"
    [    14.807] (II) Unloading int10
    [    14.807] (II) UnloadSubModule: "vbe"
    [    14.807] (II) Unloading vbe
    [    14.807] (EE) Screen(s) found, but none have a usable configuration.
    [    14.807] (EE) 
    Fatal server error:
    [    14.807] (EE) no screens found(EE) 

For extra context: this is on my MacBookPro11,5.

@aaronlevin, did you experience anything like this?

**Comment by cstrahan** [comment]
@aaronlevin: also, does yours have an AMD/Radeon graphics card?

[The Arch wiki](https://wiki.archlinux.org/index.php/MacBookPro11,x#MacBook_Pro_11.2C5) seems to suggest that some 11,5's have GeForce cards. Interestingly, that page doesn't mention AMD or Radeon at all.

**Comment by vcunat** [comment]
The live ISOs pass `nomodeset` to kernel now, IIRC. Perhaps that wasn't a good idea and only nouveau should be blacklisted?

**Comment by cstrahan** [comment]
I figured out the problem. My MacBook Pro has an AMD Radeon GPU, and without further customization, the graphical installer ISO image attempts to load the free `radeon` kernel module, but that fails because it does not support `nomodeset`, which is a kernel parameter [added by default in the ISO image](https://github.com/NixOS/nixpkgs/blob/709470f2c0bd83395f14fc7445a8a9793c0ceaa5/nixos/modules/installer/cd-dvd/iso-image.nix#L233).

[Here's a description of what `nomodeset` does](http://askubuntu.com/questions/207175/what-does-nomodeset-do):

> The newest kernels have moved the video mode setting into the kernel. So all the programming of the hardware specific clock rates and registers on the video card happen in the kernel rather than in the X driver when the X server starts.. This makes it possible to have high resolution nice looking splash (boot) screens and flicker free transitions from boot splash to login screen. Unfortunately, on some cards this doesnt work properly and you end up with a black screen. Adding the nomodeset parameter instructs the kernel to not load video drivers and use BIOS modes instead until X is loaded.

In order to start the X server, I have to wait for Gummiboot to come up, then I type `e`, which let's me edit the kernel parameters. If you scroll over to the far right, you can remove the "nomodeset" parameter. After hitting enter and waiting for NixOS to boot up, `start display-manager` will then work.

**Comment by cstrahan** [comment]
> The live ISOs pass `nomodeset` to kernel now, IIRC. Perhaps that wasn't a good idea and only nouveau should be blacklisted?

I'm not familiar enough with this territory, so I'll leave that up to someone else more experienced. I imagine we want to make the default be whatever works on most current machines, but I wouldn't know whether having keeping or removing `nomodeset` achieves that.

**Comment by vcunat** [comment]
I doubt we have an active contributor with substantial experience/knowledge about these things. What occurs to me is to find out and just copy behavior of some other distro where these things mostly work.

Maybe leaving this is safer, as without nomodeset some machines don't even show the terminal.

**Comment by CMCDragonkai** [comment]
Met this problem as well, but I have only a discrete Geforce GTX 965M. Exactly the same status log given above from `systemctl status display-manager.service`. What can be done to get atound this?

**Comment by CMCDragonkai** [comment]
Oh it works when I boot using BIOS, but it fails when I boot using EFI. How strange.

**Comment by CMCDragonkai** [comment]
Further testing has shown that as long as "Launch Video OpROM policy" is set to BIOS/Legacy mode, the `start display-manager` works. It works regardless of whether the computer is booted into UEFI or BIOS. **Only when the Video OpROM is UEFI does the `start display-manager` fail**. @cstrahan perhaps you can try to see if switching that option in the BIOS changes anything for you.

The nomodeset removal didn't work for me.

**Comment by aaronlevin** [comment]
@cstrahan hey, sorry I didn't see this until today. i didn't use the Live CD, so I didn't hit this.

Not sure what graphics card. Mine says Intel Idris Pro?

**Comment by paulkoerbitz** [comment]
I had the same issue on a MacBookPro12,1 (13", 2015 Retina) with the nixos graphical live cd (15.09.760.ba0d05c). 

The workaround indicated by @cstrahan (removing the `nomodeset` parameter on boot) fixed this for me.

I am fairly sure that I have a intel idris graphics card.

**Comment by CMCDragonkai** [comment]
`nomodeset` didn't work for me. But that's probably because I have a Nvidia graphics card. @paulkoerbitz did you boot into UEFI or BIOS?

**Comment by paulkoerbitz** [comment]
@CMCDragonkai I used UEFI.

**Comment by vrthra** [comment]
Removing `nomodeset` as suggested by @cstrahan worked for me (15" 2015 Retina). However the touchpad does not work in the X.

**Comment by Profpatsch** [comment]
(triage) what’s the status?

**Comment by Timoha** [comment]
I'm trying NixOS graphical 16.09.1066 Live CD (booting from a USB) on Macbook Pro 8.2 and still experiencing an issue when running `systemctl start display-manager` which results in just one `_` in top left corner and nothing happening.

**Comment by ashgillman** [comment]
I am having the same issue as @Timoha , can't start X from LiveCD or after installation. I have mid 2012 MBP. I have tried configurations recommended by [mbbx6ssp](https://github.com/mbbx6spp/mbp-nixos/blob/master/etc/nixos/configuration.nix), [damiancassou](https://gist.github.com/DamienCassou/e1e6b3746789276b7b25) and [puffnfresh](https://github.com/puffnfresh/nix-files/tree/master/macbook-late-2013).

Once I get to the `_` screen, I am no longer able to switch between TTY inputs.

**Comment by corazza** [comment]
Wanted to try out NixOS and was hit by this issue as well with my Nvidia 1060 card. Trying to start the display manager results in three screen flickers, and a totally uninformative log (I could try and look more closely if someone tells me where). I tried both stable and unstable, booting through UEFI, I tried to boot the USB in legacy but I couldn't find that option (secure boot disabled, UEFI **or** legacy selected in Rufus when burning the image though, but there is still no legacy USB entry in my BIOS (Gigabyte MB)).

Does anyone have any idea what to do here, I'd really like to try this OS out?

**Comment by cstrahan** [comment]
@corazza I just a similar experience (re)installing NixOS on my early 2015 MacBookPro (MacBookPro11,3) with NVidia graphics. In my case, the problem was that the integrated GPU won't work on MacBooks with dedicated GPUs, but then the open source `nouveau` driver wasn't loaded. I confirmed this by running `lsmod | grep nouveau`, which didn't give any results -- and I know that we don't bundle the proprietary driver, so that could only mean that there was no suitable driver loaded.

IIRC, the `nouveau` driver requires that `nomodeset` _not_ be present on the command line (from testing earlier, this seems correct), so I made sure to select a suitable the grub boot entry at startup, then I ran `modprobe nouveau` at the initial shell, then `systemctl start display-manager` worked as expected. During installation, I can then configure the proprietary Nvidia driver in `configuration.nix`.

I'm not really sure where the problem lies: shouldn't the kernel automatically load `nouveau` (unless explicitly blacklisted, as one would do to use the proprietary driver)? It would appear that something somewhere has regressed, because when I installed NixOS on this laptop back in 2015 there was no need to manually load `nouveau`.

Any way, hopefully that helps. I suspect people with AMD GPUs might run into the same sort of issue, in which case they may want to try running the analogous `modprobe radeon`.
